yucca/training/lightning_modules/YuccaLightningModule.py

```diff
@@ -211,14 +211,24 @@ def configure_optimizers(self):
         # Finally return the optimizer and scheduler - the loss is not returned.
         return {"optimizer": self.optim, "lr_scheduler": self.lr_scheduler}
 
-    def load_state_dict(self, *args, **kwargs):
+    def load_state_dict(self, state_dict, *args, **kwargs):
+        # First we filter out layers that have changed in size
+        # This is often the case in the output layer.
+        # If we are finetuning on a task with a different number of classes
+        # than the pretraining task, the # output channels will have changed.
+        old_params = copy.deepcopy(self.state_dict())
+        state_dict = {
+            k: v for k, v in state_dict.items() if (k in old_params) and (old_params[k].shape == state_dict[k].shape)
```

Comment:
Why do we have to this both in the lightning module _and_ in the yuccanet, which is then called below on line 230?

Reply:
In finetune - and in Lightning in general - we invoke the load_state_dict method of the LightningModule (which is a nn.module). The YuccaNet method is for cases where it's loaded outside the LightningModule. And we have to use the LightningModule method because otherwise the layer names will not match when we load them from checkpoints saved by Lightning (because they're saved as "model.conv1.weight" etc, and if you try to load them directly into a network it'll be "conv1.weight")

Reply:
I think it would make sense to gather the logic a bit instead of duplicating it. I guess the lightning module contains a net deriving from `YuccaNet` in `self.model`. Couldn't we call `self.model.load_state_dict` somehow?


Alternatively, and I really do like the above option better if it is in any way possible, but I would at least suggest a util function placed somewhere where it makes sense like

```python
def apply_state_dict_to_current(state_dict: {}, current_state_dict: {}):
    return {
            k: v
            for k, v in state_dict()
            if (k in current_state_dict) and (current_state_dict[k].shape == state_dict[k].shape)
    }
```
Which could then easily be called from both load_state_dict methods.

Reply:
First is not possible, then we again end up calling the YuccaNet state dict which has the wrong keys

We could wrap it, but I feel like that is a bit overkill for 1 line of very trivial dict comprehension and it might end up making it less clear what is happening

Reply:
Okay, LGTM then! :-)